src/api/load.js

```diff
@@ -107,6 +110,10 @@ export async function load(
   const graphJSON = WeightedGraph.toJSON(weightedGraph);
   await fs.writeFile(graphFile, stringify(graphJSON));
 
+  const pluginsFile = path.join(projectDirectory, "pluginDeclarations.json");
+  const pluginsJSON = pluginsToJSON(plugins);
+  await fs.writeFile(pluginsFile, stringify(pluginsJSON));
```

Comment:
Directly managing the project filesystem from load is going to be replaced, since `DataDirectory` / `ProjectStorageProvider` is responsible for this. https://github.com/sourcecred/sourcecred/blob/master/src/backend/dataDirectory.js

Would you be open to updating `DataDirectory` / `ProjectStorageProvider` to accept a `PluginDeclarations` type instead, and switch to `DataDirectory` for writing the file instead?

Reply:
I updated the DataDirectory and ProjectStorage classes to accept the PluginDeclarations. I didn't refactor load to actually use the DataDirectory, but it should make it trivial for you to include this change in your refactored pipeline (if this merges first), or trivial for me to incorporate it in yours (if yours merges first).